geomstats/geometry/discrete_curves.py

```diff
@@ -1060,31 +1217,29 @@ def _discrete_horizontal_geodesic_single(self, bundle, initial_point, end_point)
         k_sampling_points = bundle.total_space.k_sampling_points
         t_space = gs.linspace(0.0, 1.0, k_sampling_points)
 
-        end_point_with_origin = insert_zeros(
-            end_point, axis=-bundle.total_space.point_ndim
-        )
-        spline_end = CubicSpline(t_space, end_point_with_origin, axis=0)
-
         current_end_point = end_point
         repar_inverse_end = []
-        for _ in range(self.max_iter):
+        for index in range(self.max_iter):
             horizontal_path_with_origin = self._iterate(
                 bundle,
                 times,
                 t_space,
                 initial_point,
                 current_end_point,
                 repar_inverse_end,
-                spline_end,
+                end_spline,
```

Comment:
I'm afraid if we do that it will reduce the readability of the code (which is already difficult), whereas we won't simplify much. Or at least the way I have in mind, which may not be the right way.